### DISCUSSION TITLE
Frsky telemetry sats in degrees

### DISCUSSION THREAD

**Discussion by MichaelETHOS** [iNavFlight/inav#9621] [General]
#Hello everyone, after a long break with inav I started again.  I built a vtol and use frsky ethos with a tdmx receiver.  Now I want to understand the telemetry and I'm not getting anywhere with it.  the number of satellites are displayed in degrees ° (temp2) at Ethos.  how can I set that I can see the number of sats as a number on the x20.  or is it possible to update inav so that it displays the frsky telemetry better like Crossfire or jeti.  A friend uses jeti and he gets it displayed as sats.  thank you for your help.

**Comment by MrD-RC** [comment]
The issue is that the LUA script doesn’t exist of ETHOS. The number of sats is part of the Tmp2 (not Temp2) sensor. You can see how to understand the values in the telemetry.md document. It is not a temperature reading l, so should not be displayed in degrees. When I get a chance, I can check on OpenTX to find out the units for Tmp2 (note Tmp1 should also not be degrees). It should be a 4 digit number.

**Comment by MichaelETHOS** [comment]
for me it is in degrees.  and when I have the 6 sats it says 305° and I can arm
![IMG_20240105_114946](https://github.com/iNavFlight/inav/assets/84266308/75f34cdf-54b3-4075-a67d-3b64d982d6c1)

**Reply by MichaelETHOS** [reply]
I wrote in GitHub Ethos the Problem and this was the answer


This appears to be an iNav issue at first glance, if the sensor ID for the value indicates it is a temp sensor, it is going to get displayed as such, which means degrees not an unformatted value.

it looks like iNav is trying to stuff their data into an improper data format

**Comment by MrD-RC** [comment]
It will likely just need adding to the Telemetry document to rename and change the type on these 2 sensors. OpenTX did not rename the sensors or set them to degrees. Only ETHOS is doing this.

**Reply by MichaelETHOS** [reply]
I think inav it’s displayed in degrees because iNav is sending the data tagged as a temperature sensor.
Can't you just look at it again to see if that's the case?  
I plugged my tdmx with s.port into tx and selected sbus in inav because there are no telemetry providers for frsky. FrSky, Graupner HoTT V4, SmartPort (S.Port), LightTelemetry (LTM).  MAVLink, IBUS, Crossfire and GSM SMS.  s.port yes but not frsky a colleague has jeti and also gets sats in his telemetry

**Reply by MrD-RC** [reply]
The ID used for Tmp1 and Tmp2 is for temperature sensors. ETHOS is just using different names. **Temp. 2** compared to **Tmp2"**. I've checked OpenTX, and that is setting them to degrees C too. So it should be fine, so long as you don't change it. There could be unit conversion going on if you do. I don't know why temperature sensor IDs were used in the first place. But changing them now would break compatibility with the INAV LUA. The sensors would have a number assigned like 0421 and 0422.

I'm sure I read somewhere that you said it shows 316 when you can arm. That would make perfect sense, when you break down the sensor.
```
3 1 6
^ ^ ^
| | | Indicates you have 6 satellite fixes
| | Shows a GPS accuracy of 1, which is not great. You probably have a low HDOP, 9 is the best
| Indicates you have a GPS fix and home lock
```
How you interpret the sensor is documented in https://github.com/iNavFlight/inav/blob/master/docs/Telemetry.md#available-smartport-sport-sensors

**Reply by MichaelETHOS** [reply]
When I switch on the vtol, all the sensors are running.  All I know is that when the FC beeps, I have the minimum sats.  then the value in temp2 increases and a short time later I lost sensor from temp2.  For me the sensor id stands for temp1 1b 0400 and temp2 1b 0410.... but I heard that opentx and Ethos no longer have anything to do with each other.  Of course it's all frsky.  I have no idea about programming.  but if you always talk about opentx and inav lua then it shouldn't be a problem to create an inav lua for Ethos if the language is the same as opentx?

**Reply by MrD-RC** [reply]
Yes, they are the sensor numbers. To me it sounds like they’re working. You just need to decode Temp. 2 to get how many satellites you have. 

Losing the sensor is something else entirely. That is down to the RF link. 

The LUA programming is not the same for OpenTX and ETHOS. If it was, the OpenTX LUA would just work. A LUA script could be made for ETHOS. We would welcome that. This is an open source project. So anyone is welcome to contribute a conversion of the OpenTX LUA to ETHOS.

**Reply by MichaelETHOS** [reply]
yes, how can I do that?  that I get the sats displayed. What should I set?  and on the other.  Do you know someone who can implement writing a lua?  or do you know the right contacts who can get this started?

**Reply by MrD-RC** [reply]
I’ve told you about 3 times how you can see how many stats there are. [Here](https://github.com/iNavFlight/inav/discussions/9621#discussioncomment-8035458) for example. You don’t need to set anything. You only have to read the numbers and interpret them.

**Comment by Toolroom1** [comment]
Hi MrD-RC
First thankyou for all your super videos.
Iam running the latest versions of ETHOS and INAV and would love to know how many sats I have, I can understand I think what you discribed above with reference to what each number meaning is. My problem is Temp2 reads 0 when first switched on, then when I get a lock and the GPS led flashes it reads 5c, a few seconds later I get the report, sensor lost and reading goes Red. Temp1 stays at 2c most times I have looked at it.
Any ideas would be much appreciated.
In passing Auto Launch worked brilliantly first time, with the X20RS counting down 10 seconds once throttle raised before motor start, great stuff.

**Comment by MrD-RC** [comment]
I have written a small LUA script to try and make the Tmp2 sensor more readable. I found that it is going red after a short while too. I don't know what's causing it. Whether it's on the FrSky or INAV side. But I'll take a look when I have the time. 

I've also added something so that in the next version of INAV. It won't use the Tmp2 sensor. GNSS will have its own ID. Which will hopefully also help.

**Comment by Toolroom1** [comment]
Wow, thank you so much, would I be able to use the LUA script and if yes how can I get it.
Just wish I had your skills, bit late to learn at 66.

Many Thanks

**Comment by Toolroom1** [comment]
To help me understand, part of GPS data stream data, is the number of satalites, what I struggle with is the fact that my FPV goggles can show the number of satalites, yet ETHOS cannot, who gets to look and unpick the data stream first INAV, I guess as the GPS is connected to the controller. Also I note that although ETHOS looses the sensor, the number of sats still seems to be displayed in the goggles, as far as I can remember.

**Comment by MrD-RC** [comment]
The LUA script is available at https://github.com/MrD-RC/ETHOS-GNSS-display-for-INAV/

The OSD is directly updated by INAV. So not related to the telemetry. I'll see if the same telemetry freeze happens on OpenTX or with CRSF on ETHOS. Hopefully it can be tracked down and fixed.

**Reply by Molecuman** [reply]
Hello. I´m getting a 4 digit number on the 480 sensor with Frsky ETHOS. From what I've read in this post, the widget reads 3 digits, but iNav mentions 4 in the documents. I wish I knew how to fix it, but I have no idea. If you have a moment, could you please take a look at the widget's LUA script? Thank you so much!

**Reply by MrD-RC** [reply]
The widget reads both. ETHOS will chop off the leading zero. The widget knows and handles it. 

The LUA will not alter what you see in the sensors list. Which will be where you're seeing the 3 digit number.

**Reply by Molecuman** [reply]
I finally got it to work. For some reason, the sensor didn’t show up at first, but I added it as a DIY sensor and the widget finally came to life. Thank you so much!

**Comment by Toolroom1** [comment]
Thank you so much, the widget is on my first screen displaying sats, I find its goes from 0 sats to 4 sats (number in Green) then turns grey once the Temp2 reads 324c, hope fully this will be sorted in time, thank you again for your time, its much appreciated.

**Comment by Toolroom1** [comment]
Could the Temp2 going off (and I cant see a connection but thought I would ask) have anything to do with the YAPPU mapping widget not showing the GMcatcher maps, as myself and others have mentioned?
